### DISCUSSION TITLE
No more rebase locally

### DISCUSSION THREAD

**Discussion by shivarm** [Hyperfoil/Horreum#755] [Ideas]
Whenever we need to rebase our branch against the master or base then we need to do locally. I would say enable the button that will handle this and there will will be no more rebase locally.

#### step- Go to repo general setting and check the following option
![image](https://github.com/Hyperfoil/Horreum/assets/91419219/32bc6e8d-fcf1-4b88-b1ba-a0ad12ce967a)

Advantage- it will save our time!

@jesperpedersen

**Comment by johnaohara** [comment]
How does this handles conflicts in the rebase?

It might work well for changes where there are not conflicts, but what is the workflow for a rebase where there are conflicts?

 - does it leaves the PR branch with an active rebase that need to be resolved before anything can be pushed to/pulled from?
 - Does it give you the option to abort the rebase if there are conflicts?
 - Does it prevent the rebase from starting if GH detects that are conflicts?

If it forces you to resolve conflicts through a WebUI, I would not enable this.

But, if it prevents you from rebasing when there are conflicts, or allows you to abort a rebase if there is conflicts, then it will help in the situation where changes in main branch have been pulled in while your PR is open

**Reply by shivarm** [reply]
Totally agree with above points:)

**Comment by jesperpedersen** [comment]
I don't think we want this - people can rebase.

And, we def don't want a situation where merge commits will show up.
